datanode/networkhistory/service.go

```diff
@@ -274,14 +278,29 @@ func (d *Service) LoadNetworkHistoryIntoDatanodeWithLog(ctx context.Context, log
 
 	chunks := chunk.Slice(datanodeBlockSpan.ToHeight+1, chunk.HeightTo)
 	loadResult, err := d.snapshotService.LoadSnapshotData(ctx, log, chunks, connConfig, withIndexesAndOrderTriggers, verbose)
-	if err != nil {
+	if err == nil {
+		log.Info("loaded all available data into datanode", logging.String("result", fmt.Sprintf("%+v", loadResult)),
+			logging.Duration("time taken", time.Since(start)))
+
+		return loadResult, err
+	}
```

Comment:
Why not just put that in the for loop and give it 1 more retry?

Reply:
Because that would be the obvious thing to do :stuck_out_tongue:  fair enough

Reply:
hum, alright